### DISCUSSION TITLE
HS200 switch now showing as switch and outlet

### DISCUSSION THREAD

**Discussion by ryannowlin** [plasticrake/homebridge-tplink-smarthome#331] [General]
Have been running this plugin with home bridge for a couple years, no issues.  I have 2 HS200 switches, 2 KP405 and multiple HS105 on the network. Just upgraded to latest version of HB and now my HS200 switches show in Homekit as both a switch and an outlet, and the switch icon does nothing - have to use the outlet to toggle on/off.    Had to reprogram all my automations to reference the outlet instead of the switch.  The other kasa devices are working as expected.   

Anyone else experience this?  Suggestions on how to adjust config to fix?

thanks!
Ryan

![hs200 switch](https://github.com/plasticrake/homebridge-tplink-smarthome/assets/102236298/1532f94d-140f-45f5-b596-3531b6553685)

**Comment by jasonlboggs** [comment]
There is a bug open for this.  revert back to the latest 7.x version, 8.x is not ready for prime time.
